mlflow/pyfunc/model.py

```diff
@@ -583,7 +610,14 @@ def _convert_input(self, model_input):
             elif isinstance(model_input, list) and all(isinstance(x, dict) for x in model_input):
                 keys = [x.name for x in self.signature.inputs]
                 return [{k: d[k] for k in keys} for d in model_input]
-
+        elif isinstance(hints.input, type) and (
+            issubclass(hints.input, ChatCompletionRequest)
+            or issubclass(hints.input, MultiturnChatRequest)
+        ):
+            # If the type hint is a RAG dataclass, we hydrate it
+            if isinstance(model_input, pd.DataFrame):
+                # Since single input is expected, we take the first row
```

Comment:
I think we should throw if there are multiple rows, which would indicate invalid input